chore: Release/1.2.0 fixes back to develop

ShadauxCat opened the pull request.

ashwinimurt committed:
chore: Bump BGO version to 1.2.0

ShadauxCat committed:
fix: Fixes for Deferred Messaging Tests [MTT-5102] (#2319)

NoelStephensUnity commented on com.unity.netcode.gameobjects/CHANGELOG.md:
```diff
@@ -6,7 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 Additional documentation and release notes are available at [Multiplayer Documentation](https://docs-multiplayer.unity3d.com).
 
-## [Unreleased]
+## [1.2.0] - 2022-11-21
```
maybe exclude the changelog?

NoelStephensUnity commented on com.unity.netcode.gameobjects/package.json:
```diff
@@ -2,7 +2,7 @@
     "displayName": "Netcode for GameObjects",
     "description": "Netcode for GameObjects is a high-level netcode SDK that provides networking capabilities to GameObject/MonoBehaviour workflows within Unity and sits on top of underlying transport layer.",
-    "version": "1.1.0",
+    "version": "1.2.0",
```
Probably want to exclude this as well.

NoelStephensUnity reviewed (approved):
If we are clear to merge the changelog updates and the version update then the rest looks good to me!

NoelStephensUnity committed:
chore: migrating PR-2320 (FindObjectsByType) into v1.2.0 release branch (#2325)

fix: This fixes the issue with recent 2023 update where FindObjectsOfType was renamed to FindObjectsByType (#2320)

* update
Adding the 2023.1.a21 FindObjectsByType fix for NGO using the UNITY_2023_1_OR_NEWER define.
Modified a few of the FindObjectsSortMode for areas that looked like they might require their results ordered by InstanceID.

ashwinimurt committed:
chore: Ignore BasicBuildTest on release/1.2.0 (#2339)

fluong6 committed:
fix: fix UnityTransport_RestartSucceedsAfterFailure test for release 1.2.0 (#2349)

fixing failing utp test case for release 1.2
